### PR TITLE
podio: add gcc14 conflict for podio

### PR DESCRIPTION
**Pull request by joequant**
add a gcc14 conflict for older version of podio

### CODE REVIEW

File: var/spack/repos/builtin/packages/podio/package.py [hunk 1]
```diff
@@ -105,6 +105,7 @@ class Podio(CMakePackage):
     depends_on("py-tabulate", type=("run", "test"), when="@0.16.6:")
 
     conflicts("+rntuple", when="@:0.16", msg="rntuple support requires at least podio@0.17")
+    conflicts("%gcc@14", when="@:0.99")
```
Comment (tmadlener) [inline var/spack/repos/builtin/packages/podio/package.py]: Can we add a reason for why it fails to compile via `msg`?

Comment (joequant) [reply var/spack/repos/builtin/packages/podio/package.py]: added message for why it files to compile.  0.99 is missing std include files.  These would be easy to add via a patch but the unpatched version will not compile.

**Comment by tmadlener** [comment]
This patch doesn't do the trick already? 

https://github.com/spack/spack/blob/02320b18f32e9119e83ba620bf19a2bd8265564d/var/spack/repos/builtin/packages/podio/package.py#L110-L114

It is the only commit in podio that I could find that mentions gcc14 explicitly.

**Comment by jmcarcell** [comment]
Yes, this should be closed, it was done in https://github.com/spack/spack/pull/45854.

**Comment by vvolkl** [comment]
Closing as fixed in https://github.com/spack/spack/pull/45854

**Pull request closed without merging**